### ISSUE TITLE
Pulse Snapshot command should align with circuit Snapshot

### ISSUE THREAD

**Issue by taalexander** [Qiskit/qiskit-terra#2592]
### What is the expected enhancement?
The pulse `Snapshot` command should align with the circuit `Snapshot`. This means:

`name` should be changed to `label`. As all `Instruction`s may have a `name` this can be made an optional `kwarg`. If `name` is not supplied it should default to `label`.

`snap_type` should be changed too `snap_label` and it shoud be made a kwarg with default value `statevector`.

As there is currently no simulator, I believe it is safe to make these changes without a backwards compatibility shim.

**Comment by chowington** [comment]
Hi taalexander, new to the community. I found the pulse Snapshot object, but where is the circuit Snapshot object you're referring to? Just for reference.

**Comment by taalexander** [comment]
That is the one 😄. Welcome to the community! Please feel free to access any additional question you may have.

**Comment by chowington** [comment]
Looks like the `snap_type` equivalent in `extensions.simulator` is `snapshot_type`. Is that what you mean instead of `snap_label`?

If so, here's the proposed `__init__`:
```
    def __init__(self, label: str, snapshot_type: str = 'statevector', name: str = None):
        """Create new snapshot command.

        Args:
            label (str): Snapshot label which is used to identify the snapshot in the output.
            snapshot_type (str): Type of snapshot, e.g., “state” (take snapshot of the quantum state).
                The types of snapshots offered are defined in a separate specification
                document for simulators.
            name (str): Snapshot name which defaults to label, but can be different than label.
        """
        self._type = snapshot_type
        self._channel = SnapshotChannel()
        Command.__init__(self, duration=0)
        self._label = Snapshot.create_name(label)

        if name is not None:
            self._name = Snapshot.create_name(name)
        else:
            self._name = self._label

        Instruction.__init__(self, self, self._channel, name=self.name)
        self._buffer = 0
```

I'd also add the (new) name param to `__eq__` and `__repr__`:
```
    def __eq__(self, other):
        """Two Snapshots are the same if they are of the same type
        and have the same label, type, and name.

        Args:
            other (Snapshot): other Snapshot,

        Returns:
            bool: are self and other equal.
        """
        if (type(self) is type(other) and
                self.label == other.label and
                self.type == other.type and
                self.name == other.name):
            return True
        return False
```

```
    def __repr__(self):
        return '%s(%s, %s, %s) -> %s' % (self.__class__.__name__, self.label,
                                         self.type, self.name, self.channels)
```

I'd also add a `@property label` function to match.

**Comment by taalexander** [comment]
This looks great @chowington, thanks for taking this on!

**Comment by chowington** [comment]
No problem! I think I'll not print `name` in `__repr__` if it's the same as `label`, as that's still a valid definition and is clearer to the user who didn't provide `name`:
```
>>> Snapshot('label')
Snapshot(label, statevector) -> (SnapshotChannel(0),)
>>> Snapshot('label', name='name')
Snapshot(label, statevector, name) -> (SnapshotChannel(0),)
```
I'll get a PR up!

**Comment by taalexander** [comment]
Closed by #2671.
